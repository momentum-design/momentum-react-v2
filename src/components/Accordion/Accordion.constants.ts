const CLASS_PREFIX = 'md-accordion';

const DEFAULTS = {
  DEFAULT_EXPANDED: true,
};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
  headerRow: `${CLASS_PREFIX}-header-row`,
  headerHeading: `${CLASS_PREFIX}-header-heading`,
  headerButton: `${CLASS_PREFIX}-header-button`,
};

export { CLASS_PREFIX, DEFAULTS, STYLE };
