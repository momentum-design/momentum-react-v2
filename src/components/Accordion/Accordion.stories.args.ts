import { commonStyles } from '../../storybook/helper.stories.argtypes';

const accordionArgTypes = {
  children: {
    description: 'Content of the accordion panel',
    control: { type: 'text' },
    table: {
      type: {
        summary: 'ReactNode',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  heading: {
    description: 'Content of the heading button',
    control: { type: 'text' },
    table: {
      type: {
        summary: 'ReactNode',
      },
    },
  },
  headingRightContent: {
    description: 'Content on the right of the heading button',
    control: { type: 'text' },
    table: {
      type: {
        summary: 'ReactNode',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  defaultExpanded: {
    description: 'Whether if the accordion should be expanded by default',
    control: { type: 'boolean' },
    table: {
      type: {
        summary: 'boolean',
      },
      defaultValue: {
        summary: 'true',
      },
    },
  },
};

export { accordionArgTypes };

export default {
  ...commonStyles,
  ...accordionArgTypes,
};
