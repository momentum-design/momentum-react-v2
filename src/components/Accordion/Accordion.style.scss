.md-accordion-wrapper {
  .md-accordion-header-row {
    display: grid;
    grid-template-columns: 1fr max-content;
    align-items: center;
    gap: 0.5rem;
  }

  .md-accordion-header-heading {
    width: 100%;
  }

  .md-accordion-header-button {
    display: flex;
    gap: 0.5rem;
    align-items: center;
    background: none;
    border: none;
    justify-content: flex-start;
  }
}
