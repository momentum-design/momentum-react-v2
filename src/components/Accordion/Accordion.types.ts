import { CSSProperties, ReactNode } from 'react';
import { ButtonSimpleProps } from '../ButtonSimple';

type ButtonSimpleCustomPropKeys =
  | 'className'
  | 'id'
  | 'aria-expanded'
  | 'aria-controls'
  | 'onPress';

// props applied on the button, defined by the accordion component
export type ButtonCustomProps = Required<Pick<ButtonSimpleProps, ButtonSimpleCustomPropKeys>> & {
  'data-interactive': boolean;
  'data-shape': string;
  'data-padded': boolean;
};

// additional props applied on the button, user-specified, not allowed to overlap with accordion-defined props
export type ButtonAllowedProps = Omit<ButtonSimpleProps, ButtonSimpleCustomPropKeys>;

export interface Props {
  /**
   * Child components of this Accordion.
   */
  children?: ReactNode;

  /**
   * Custom class for overriding this component's CSS.
   */
  className?: string;

  /**
   * Custom id for overriding this component's CSS.
   */
  id?: string;

  /**
   * Custom style for overriding this component's CSS.
   */
  style?: CSSProperties;

  /**
   * Element to render in the header area.
   */
  heading: ReactNode;

  /**
   * Element to render in the header area, aligned to the right.
   */
  headingRightContent?: ReactNode;

  /**
   * Whether if the accordion should be expanded by default.
   */
  defaultExpanded?: boolean;

  /**
   * Hierarchical level of the accordion within a structure.
   */
  ariaLevel: number;

  /**
   * Extra props to be added to the header button.
   */
  buttonProps?: ButtonAllowedProps;
}
