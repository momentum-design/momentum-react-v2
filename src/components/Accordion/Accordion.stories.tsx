import { MultiTemplate, Template } from '../../storybook/helper.stories.templates';
import { DocumentationPage } from '../../storybook/helper.stories.docs';
import StyleDocs from '../../storybook/docs.stories.style.mdx';

import Accordion, { AccordionProps } from './';
import argTypes from './Accordion.stories.args';
import Documentation from './Accordion.stories.docs.mdx';

export default {
  title: 'Momentum UI/Accordion',
  component: Accordion,
  parameters: {
    expanded: true,
    docs: {
      page: DocumentationPage(Documentation, StyleDocs),
    },
  },
};

const Example = Template<AccordionProps>(Accordion).bind({});

Example.argTypes = { ...argTypes };

Example.args = {
  children: 'Example',
  heading: 'Accordion title',
};

const Common = MultiTemplate<AccordionProps>(Accordion).bind({});

Common.argTypes = { ...argTypes };

Common.args = {
  children: 'Example',
};

Common.parameters = {
  variants: [
    { children: undefined, heading: 'Accordion with no content' },
    { children: 'Example B', heading: 'Accordion title' },
    {
      children: 'Example C',
      heading: 'Accordion with stuff on the right',
      headingRightContent: 'Some stuff here',
    },
  ],
};

export { Example, Common };
