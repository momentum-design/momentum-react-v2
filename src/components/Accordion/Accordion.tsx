import React, { FC, forwardRef, useMemo, useState } from 'react';
import classnames from 'classnames';

import { STYLE, DEFAULTS } from './Accordion.constants';
import {
  STYLE as LIST_ITEM_BASE_STYLE,
  SHAPES as LIST_ITEM_BASE_SHAPES,
} from '../ListItemBase/ListItemBase.constants';
import type { ButtonCustomProps, Props } from './Accordion.types';
import './Accordion.style.scss';
import Icon from '../Icon';
import { v4 as uuidV4 } from 'uuid';
import Text from '../Text';
import ButtonSimple from '../ButtonSimple';

const Accordion: FC<Props> = forwardRef<HTMLButtonElement, Props>((props, providedRef) => {
  const {
    children,
    className,
    id,
    style,
    heading,
    headingRightContent,
    defaultExpanded = DEFAULTS.DEFAULT_EXPANDED,
    ariaLevel,
    buttonProps,
  } = props;

  const [expanded, setExpanded] = useState(defaultExpanded);
  const accordionId = useMemo(() => id || uuidV4(), [id]);

  const headerButtonId = `${accordionId}-header-button`;
  const panelId = `${accordionId}-panel`;

  const onClick = () => {
    setExpanded(!expanded);
  };

  const buttonCustomProps: ButtonCustomProps = {
    className: classnames(STYLE.headerButton, LIST_ITEM_BASE_STYLE.wrapper),
    'data-interactive': true,
    'data-shape': LIST_ITEM_BASE_SHAPES.isPilled,
    'data-padded': true,
    id: headerButtonId,
    'aria-expanded': expanded,
    'aria-controls': panelId,
    onPress: onClick,
    ref: providedRef,
  };

  return (
    <div className={classnames(className, STYLE.wrapper)} id={accordionId} style={style}>
      <div className={STYLE.headerRow}>
        <div className={STYLE.headerHeading} role="heading" aria-level={ariaLevel}>
          <ButtonSimple {...buttonCustomProps} {...buttonProps}>
            <Icon name={expanded ? 'arrow-down' : 'arrow-right'} scale={12} />
            {typeof heading === 'string' ? <Text type="body-secondary">{heading}</Text> : heading}
          </ButtonSimple>
        </div>
        {headingRightContent}
      </div>
      {expanded ? (
        <div role="region" aria-labelledby={headerButtonId} id={panelId}>
          {children}
        </div>
      ) : null}
    </div>
  );
});

export default Accordion;
