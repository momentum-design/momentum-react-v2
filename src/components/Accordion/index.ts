import { default as Accordion } from './Accordion';
import * as CONSTANTS from './Accordion.constants';
import { Props } from './Accordion.types';

export { CONSTANTS as ACCORDION_CONSTANTS };

export type AccordionProps = Props;

export default Accordion;
