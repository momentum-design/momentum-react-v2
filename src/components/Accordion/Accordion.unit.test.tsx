import React from 'react';
import { mount } from 'enzyme';

import Accordion, { ACCORDION_CONSTANTS as CONSTANTS } from './';
import type { Props } from './Accordion.types';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('<Accordion />', () => {
  const defaultProps: Props = {
    heading: 'Heading text',
    ariaLevel: 5,
    id: 'accordion-id', // prop is optional, but needed in tests so snapshots stay consistent
  };

  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      const container = mount(<Accordion {...defaultProps} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = mount(<Accordion {...defaultProps} className={className} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = mount(<Accordion {...defaultProps} id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = mount(<Accordion {...defaultProps} style={style} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with headingRightContent', () => {
      expect.assertions(1);

      const headingRightContent = 'Some content on the right';

      const container = mount(
        <Accordion {...defaultProps} headingRightContent={headingRightContent} />
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', () => {
      expect.assertions(1);

      const element = mount(<Accordion {...defaultProps} />)
        .find(Accordion)
        .getDOMNode();

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', () => {
      expect.assertions(1);

      const className = 'example-class';

      const element = mount(<Accordion {...defaultProps} className={className} />)
        .find(Accordion)
        .getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', () => {
      expect.assertions(1);

      const id = 'example-id';

      const element = mount(<Accordion {...defaultProps} id={id} />)
        .find(Accordion)
        .getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const element = mount(<Accordion {...defaultProps} style={style} />)
        .find(Accordion)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    it('should populate the ids (default) and their references correctly', () => {
      const component = mount(<Accordion {...defaultProps} id="undefined" />).find(Accordion);

      const button = component.find('button');
      const panel = component.find('div[role="region"]');

      expect(button.prop('id')).toEqual(expect.any(String));
      expect(panel.prop('id')).toEqual(expect.any(String));
      expect(button.prop('aria-controls')).toBe(panel.prop('id'));
      expect(panel.prop('aria-labelledby')).toBe(button.prop('id'));
    });

    it('should populate the ids (custom) and their references correctly', () => {
      const component = mount(<Accordion {...defaultProps} id="custom-id" />).find(Accordion);

      const button = component.find('button');
      const panel = component.find('div[role="region"]');

      expect(button.prop('id')).toBe('custom-id-header-button');
      expect(panel.prop('id')).toBe('custom-id-panel');
      expect(button.prop('aria-controls')).toBe(panel.prop('id'));
      expect(panel.prop('aria-labelledby')).toBe(button.prop('id'));
    });

    it('should be expanded by default', () => {
      const component = mount(<Accordion {...defaultProps} />).find(Accordion);

      expect(component.find('button').prop('aria-expanded')).toBe(true);
      expect(component.find('div[role="region"]').exists()).toBe(true);
      expect(component.find('Icon').props()).toStrictEqual({
        name: 'arrow-down',
        scale: 12,
      });
    });

    it('should be collapsed if defaultExpanded is false', () => {
      const component = mount(<Accordion {...defaultProps} defaultExpanded={false} />).find(
        Accordion
      );

      expect(component.find('button').prop('aria-expanded')).toBe(false);
      expect(component.find('div[role="region"]').exists()).toBe(false);
      expect(component.find('Icon').props()).toStrictEqual({
        name: 'arrow-right',
        scale: 12,
      });
    });
  });

  describe('actions', () => {
    it('should toggle between expanded and not when the header is clicked on', async () => {
      const user = userEvent.setup();
      render(<Accordion {...defaultProps} />);

      const button = screen.getByRole('button');

      // initial: open
      expect(button.getAttribute('aria-expanded')).toBe('true');
      expect(screen.queryByRole('region')).not.toBeNull();

      await user.click(button);

      // after click: close
      expect(button.getAttribute('aria-expanded')).toBe('false');
      expect(screen.queryByRole('region')).toBeNull();

      await user.click(button);

      // after click: open
      expect(button.getAttribute('aria-expanded')).toBe('true');
      expect(screen.queryByRole('region')).not.toBeNull();

      await user.click(button);

      // after click: close
      expect(button.getAttribute('aria-expanded')).toBe('false');
      expect(screen.queryByRole('region')).toBeNull();
    });

    it('should toggle between expanded and not when an "enter" keydown event is sent to the header', async () => {
      const user = userEvent.setup();
      render(<Accordion {...defaultProps} />);

      const button = screen.getByRole('button');

      button.focus();

      // initial: open
      expect(button.getAttribute('aria-expanded')).toBe('true');
      expect(screen.queryByRole('region')).not.toBeNull();

      await user.keyboard('[Enter]');

      // after click: close
      expect(button.getAttribute('aria-expanded')).toBe('false');
      expect(screen.queryByRole('region')).toBeNull();

      await user.keyboard('[Enter]');

      // after click: open
      expect(button.getAttribute('aria-expanded')).toBe('true');
      expect(screen.queryByRole('region')).not.toBeNull();

      await user.keyboard('[Enter]');

      // after click: close
      expect(button.getAttribute('aria-expanded')).toBe('false');
      expect(screen.queryByRole('region')).toBeNull();
    });
  });
});
