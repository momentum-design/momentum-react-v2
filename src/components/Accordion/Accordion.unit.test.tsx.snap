// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> snapshot should match snapshot 1`] = `
<ForwardRef
  ariaLevel={5}
  heading="Heading text"
  id="accordion-id"
>
  <div
    className="md-accordion-wrapper"
    id="accordion-id"
  >
    <div
      className="md-accordion-header-row"
    >
      <div
        aria-level={5}
        className="md-accordion-header-heading"
        role="heading"
      >
        <ButtonSimple
          aria-controls="accordion-id-panel"
          aria-expanded={true}
          className="md-accordion-header-button md-list-item-base-wrapper"
          data-interactive={true}
          data-padded={true}
          data-shape="isPilled"
          id="accordion-id-header-button"
          onPress={[Function]}
        >
          <FocusRing>
            <FocusRing
              focusClass="md-focus-ring-wrapper children"
              focusRingClass="md-focus-ring-wrapper children"
            >
              <button
                aria-controls="accordion-id-panel"
                aria-expanded={true}
                className="md-accordion-header-button md-list-item-base-wrapper md-button-simple-wrapper"
                data-interactive={true}
                data-padded={true}
                data-shape="isPilled"
                id="accordion-id-header-button"
                onBlur={[Function]}
                onClick={[Function]}
                onDragStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                type="button"
              >
                <Icon
                  name="arrow-down"
                  scale={12}
                />
                <Text
                  type="body-secondary"
                >
                  <small
                    className="md-text-wrapper"
                    data-type="body-secondary"
                  >
                    Heading text
                  </small>
                </Text>
              </button>
            </FocusRing>
          </FocusRing>
        </ButtonSimple>
      </div>
    </div>
    <div
      aria-labelledby="accordion-id-header-button"
      id="accordion-id-panel"
      role="region"
    />
  </div>
</ForwardRef>
`;

exports[`<Accordion /> snapshot should match snapshot with className 1`] = `
<ForwardRef
  ariaLevel={5}
  className="example-class"
  heading="Heading text"
  id="accordion-id"
>
  <div
    className="example-class md-accordion-wrapper"
    id="accordion-id"
  >
    <div
      className="md-accordion-header-row"
    >
      <div
        aria-level={5}
        className="md-accordion-header-heading"
        role="heading"
      >
        <ButtonSimple
          aria-controls="accordion-id-panel"
          aria-expanded={true}
          className="md-accordion-header-button md-list-item-base-wrapper"
          data-interactive={true}
          data-padded={true}
          data-shape="isPilled"
          id="accordion-id-header-button"
          onPress={[Function]}
        >
          <FocusRing>
            <FocusRing
              focusClass="md-focus-ring-wrapper children"
              focusRingClass="md-focus-ring-wrapper children"
            >
              <button
                aria-controls="accordion-id-panel"
                aria-expanded={true}
                className="md-accordion-header-button md-list-item-base-wrapper md-button-simple-wrapper"
                data-interactive={true}
                data-padded={true}
                data-shape="isPilled"
                id="accordion-id-header-button"
                onBlur={[Function]}
                onClick={[Function]}
                onDragStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                type="button"
              >
                <Icon
                  name="arrow-down"
                  scale={12}
                />
                <Text
                  type="body-secondary"
                >
                  <small
                    className="md-text-wrapper"
                    data-type="body-secondary"
                  >
                    Heading text
                  </small>
                </Text>
              </button>
            </FocusRing>
          </FocusRing>
        </ButtonSimple>
      </div>
    </div>
    <div
      aria-labelledby="accordion-id-header-button"
      id="accordion-id-panel"
      role="region"
    />
  </div>
</ForwardRef>
`;

exports[`<Accordion /> snapshot should match snapshot with headingRightContent 1`] = `
<ForwardRef
  ariaLevel={5}
  heading="Heading text"
  headingRightContent="Some content on the right"
  id="accordion-id"
>
  <div
    className="md-accordion-wrapper"
    id="accordion-id"
  >
    <div
      className="md-accordion-header-row"
    >
      <div
        aria-level={5}
        className="md-accordion-header-heading"
        role="heading"
      >
        <ButtonSimple
          aria-controls="accordion-id-panel"
          aria-expanded={true}
          className="md-accordion-header-button md-list-item-base-wrapper"
          data-interactive={true}
          data-padded={true}
          data-shape="isPilled"
          id="accordion-id-header-button"
          onPress={[Function]}
        >
          <FocusRing>
            <FocusRing
              focusClass="md-focus-ring-wrapper children"
              focusRingClass="md-focus-ring-wrapper children"
            >
              <button
                aria-controls="accordion-id-panel"
                aria-expanded={true}
                className="md-accordion-header-button md-list-item-base-wrapper md-button-simple-wrapper"
                data-interactive={true}
                data-padded={true}
                data-shape="isPilled"
                id="accordion-id-header-button"
                onBlur={[Function]}
                onClick={[Function]}
                onDragStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                type="button"
              >
                <Icon
                  name="arrow-down"
                  scale={12}
                />
                <Text
                  type="body-secondary"
                >
                  <small
                    className="md-text-wrapper"
                    data-type="body-secondary"
                  >
                    Heading text
                  </small>
                </Text>
              </button>
            </FocusRing>
          </FocusRing>
        </ButtonSimple>
      </div>
      Some content on the right
    </div>
    <div
      aria-labelledby="accordion-id-header-button"
      id="accordion-id-panel"
      role="region"
    />
  </div>
</ForwardRef>
`;

exports[`<Accordion /> snapshot should match snapshot with id 1`] = `
<ForwardRef
  ariaLevel={5}
  heading="Heading text"
  id="example-id"
>
  <div
    className="md-accordion-wrapper"
    id="example-id"
  >
    <div
      className="md-accordion-header-row"
    >
      <div
        aria-level={5}
        className="md-accordion-header-heading"
        role="heading"
      >
        <ButtonSimple
          aria-controls="example-id-panel"
          aria-expanded={true}
          className="md-accordion-header-button md-list-item-base-wrapper"
          data-interactive={true}
          data-padded={true}
          data-shape="isPilled"
          id="example-id-header-button"
          onPress={[Function]}
        >
          <FocusRing>
            <FocusRing
              focusClass="md-focus-ring-wrapper children"
              focusRingClass="md-focus-ring-wrapper children"
            >
              <button
                aria-controls="example-id-panel"
                aria-expanded={true}
                className="md-accordion-header-button md-list-item-base-wrapper md-button-simple-wrapper"
                data-interactive={true}
                data-padded={true}
                data-shape="isPilled"
                id="example-id-header-button"
                onBlur={[Function]}
                onClick={[Function]}
                onDragStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                type="button"
              >
                <Icon
                  name="arrow-down"
                  scale={12}
                />
                <Text
                  type="body-secondary"
                >
                  <small
                    className="md-text-wrapper"
                    data-type="body-secondary"
                  >
                    Heading text
                  </small>
                </Text>
              </button>
            </FocusRing>
          </FocusRing>
        </ButtonSimple>
      </div>
    </div>
    <div
      aria-labelledby="example-id-header-button"
      id="example-id-panel"
      role="region"
    />
  </div>
</ForwardRef>
`;

exports[`<Accordion /> snapshot should match snapshot with style 1`] = `
<ForwardRef
  ariaLevel={5}
  heading="Heading text"
  id="accordion-id"
  style={
    {
      "color": "pink",
    }
  }
>
  <div
    className="md-accordion-wrapper"
    id="accordion-id"
    style={
      {
        "color": "pink",
      }
    }
  >
    <div
      className="md-accordion-header-row"
    >
      <div
        aria-level={5}
        className="md-accordion-header-heading"
        role="heading"
      >
        <ButtonSimple
          aria-controls="accordion-id-panel"
          aria-expanded={true}
          className="md-accordion-header-button md-list-item-base-wrapper"
          data-interactive={true}
          data-padded={true}
          data-shape="isPilled"
          id="accordion-id-header-button"
          onPress={[Function]}
        >
          <FocusRing>
            <FocusRing
              focusClass="md-focus-ring-wrapper children"
              focusRingClass="md-focus-ring-wrapper children"
            >
              <button
                aria-controls="accordion-id-panel"
                aria-expanded={true}
                className="md-accordion-header-button md-list-item-base-wrapper md-button-simple-wrapper"
                data-interactive={true}
                data-padded={true}
                data-shape="isPilled"
                id="accordion-id-header-button"
                onBlur={[Function]}
                onClick={[Function]}
                onDragStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                type="button"
              >
                <Icon
                  name="arrow-down"
                  scale={12}
                />
                <Text
                  type="body-secondary"
                >
                  <small
                    className="md-text-wrapper"
                    data-type="body-secondary"
                  >
                    Heading text
                  </small>
                </Text>
              </button>
            </FocusRing>
          </FocusRing>
        </ButtonSimple>
      </div>
    </div>
    <div
      aria-labelledby="accordion-id-header-button"
      id="accordion-id-panel"
      role="region"
    />
  </div>
</ForwardRef>
`;
